crates/exex/src/model/payload.rs

```diff
@@ -598,4 +594,499 @@ mod tests {
+                .unwrap();
+
+        // Seal the header with the block hash
+        let sealed_header = SealedHeader::new(header, block_hash);
+
+        // Generate some transaction input bytes
+        let transaction_data = Bytes::from_static(&hex!("000000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"));
+
+        // Create a transaction with a signature but no hash
+        let transaction_no_hash = TransactionSignedNoHash {
+            transaction: Transaction::Legacy(TxLegacy {
+                chain_id: None,
+                nonce: 0x00,
+                gas_price: 0x0a,
+                gas_limit: 0x000f_4240,
+                to: revm_primitives::TxKind::Call(
+                    Address::from_str("0x000000000000000000000000000000000000c0de").unwrap(),
+                ),
+                value: U256::from(0x00),
+                input: transaction_data,
+            }),
+            signature: PrimitiveSignature::from_scalars_and_parity(
+                B256::from_str(
+                    "0xf225c2292ba248fe3ed544f7d45dd4172337ba41dc480c3b17af63e03d281daf",
+                )
+                .unwrap(),
+                B256::from_str(
+                    "0x35360ae92ae767c1d0a9e0358e4398174b10eeea046bceedf323e7bf3b17c652",
+                )
+                .unwrap(),
+                Parity::try_from(0x1c_u64).unwrap().y_parity(),
+            ),
+        };
+
+        // Add the transaction hash
+        let transaction = transaction_no_hash.with_hash();
+
+        // Seal the block
+        let sealed_block = SealedBlock {
+            header: sealed_header,
+            body: BlockBody {
+                transactions: vec![transaction.clone(), transaction],
+                ommers: vec![],
+                withdrawals: Some(vec![].into()),
+            },
+        };
+
+        // Transform the sealed block into a KethBlock
+        let keth_block: KethBlock = sealed_block.into();
+
+        // Transform the KethBlock into a KethPayload
+        let keth_payload: KethPayload = keth_block.clone().into();
+
+        // Create a new instance of the VirtualMachine.
+        let mut vm = VirtualMachine::new(false);
+
+        // Call gen_arg to encode the block payload and write it to the VM's memory.
+        let result = keth_payload.gen_arg(&mut vm);
+
+        // Ensure that the gen_arg call succeeded and returned a valid pointer.
+        assert!(result.is_ok(), "Expected gen_arg to succeed");
+
+        // Retrieve the resulting pointer to the memory segment for the complex Nested payload.
+        let ptr = result.unwrap();
+
+        // ############################ Pointers in header ############################
+
+        // Segment 0
+        // Withdrawal root U256 pointer:
+        // - Low
+        // - High
+        assert_eq!(
```

Comment:
you need to use the `let ptr = result.expect(...)` here as a base and offset it with `+ i` instead of creating new `Relocatable`

Reply:
`ptr` is related to the last segment (11) as pointed out here https://github.com/kkrt-labs/keth/pull/127/files#diff-59079e080003f4d5faea22b4e42a733a8eec0c581ac157d23989dfa38b668237R1090 so this will not work for the segment you mention here (0). Of course we can do this but I think that sometimes it can complicates loops and everything. Here we have a few items so that we can check by hand if everything is correct to be really sure. For me this test is really not a "computer science" test but a "construct your segment by hand" test.

Reply:
fixed